pkg/ccr/job.go

```diff
@@ -3034,14 +3035,17 @@ func (j *Job) isModifyTableColumnsCommitted(record *record.ModifyTableAddOrDropC
 		return false, err
 	}
 
-	columns, err := j.destMeta.DescribeTable(tableName)
+	columns, err := j.destMeta.DescribeTableAll(tableName)
 	if err != nil {
 		return false, err
 	}
 
+	log.Debugf("desc all table: %v", columns)
+
 	destColumnMap := make(map[string]struct{})
 	for _, col := range columns {
-		destColumnMap[col.Name] = struct{}{}
+		name := col.ColumnDesc[col.IndexName].Name
```

Comment:
Only the base index is considered here.

Take base materialized index desc:

```
materializedIndexes, err := j.destMeta.DescribeTableAll(tableName)
var baseIndex *MaterializedIndexDesc
for i, index := range materializedIndexes {
	if index.IndexName == tableName {
		baseIndex = &materializedIndexes[i]
		break
	}
}
```

Or you could simple returns a `map[string]MaterializedIndexDesc` in `DescribeTableAll`.